src/main/java/uk/gov/companieshouse/accountsDates/AccountsDates.java

```diff
@@ -0,0 +1,199 @@
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AccountsDates {
+
+	private static final String PERIOD_START = "periodStart";
+	private static final String PERIOD_END = "periodEnd";
+	private static final String DATE_FORMAT_YYYYMMDD = "yyyy-MM-dd";
+	private static final String DATE_FORMAT_D_MMMM_YYYY = "d MMMM yyyy";
+
+	public LocalDate convertStringToDate(String stringDate) {
+		DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT_YYYYMMDD);
+		return LocalDate.parse(stringDate, formatter);
+	}
+
+	public String convertDateToString(LocalDate date) {
+		DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT_YYYYMMDD);
+		return date.format(formatter);
+	}
+
+	// from 2017-03-05 to 3 March 2005
+	public String convertStringToDisplayDate(String stringDate) {
+
+		DateTimeFormatter dateFormatOriginal = DateTimeFormatter.ofPattern(DATE_FORMAT_YYYYMMDD);
+		DateTimeFormatter dateFormatDesired = DateTimeFormatter.ofPattern(DATE_FORMAT_D_MMMM_YYYY);
+		LocalDate date = LocalDate.parse(stringDate, dateFormatOriginal);
+
+		return date.format(dateFormatDesired);
+	}
+
+	public Map<String, String> getDateAndTime(String dateString) {
+		Map<String, String> timeObject = new HashMap<>();
+
+		Instant instant = Instant.parse(dateString);
+
+		LocalDateTime localDateTime = LocalDateTime.ofInstant(instant, ZoneId.of("GMT"));
+		LocalDate localDate = localDateTime.toLocalDate();
+
+		DateTimeFormatter dateFormatDesired = DateTimeFormatter.ofPattern(DATE_FORMAT_D_MMMM_YYYY);
+		DateTimeFormatter timeFormatDesired = DateTimeFormatter.ofPattern("h:mm a");
+
+		String date = localDate.format(dateFormatDesired);
+		String time = localDateTime.format(timeFormatDesired).toLowerCase();
+
+		timeObject.put("date", date);
+		timeObject.put("time", time);
+
+		return timeObject;
+	}
+	
+	/**
+	 * @param periodStart
+	 * @param periodEnd
+	 * @param isSameYear
+	 * @return
+	 * 
+	 * 	Generate balance sheet header string based on 
+	 *  period start and end dates
+	 * 
+	 */
+	public String generateBalanceSheetHeading(String periodStartString, String periodEndString, boolean isSameYear) {
+
+		Instant instantPeriodStart = Instant.parse(periodStartString);
+		Instant instantPeriodEnd = Instant.parse(periodEndString);
+
+		LocalDateTime localDateTimePeriodStart = LocalDateTime.ofInstant(instantPeriodStart, ZoneId.of("GMT"));
+		LocalDate localDatePeriodStart = localDateTimePeriodStart.toLocalDate();
+
+		LocalDateTime localDateTimePeriodEnd = LocalDateTime.ofInstant(instantPeriodEnd, ZoneId.of("GMT"));
+		LocalDate localDatePeriodEnd = localDateTimePeriodEnd.toLocalDate();
+
+		Map<String, String> resultdates = calculatePeriodRange(localDatePeriodStart, localDatePeriodEnd, isSameYear);
+
+		if (resultdates.get("periodStart") == null) {
```

Comment:
You could check if the key exists instead of storing a null in the map:
```java
if(!resultdates.hasKey("periodStart")) {
   return resultdates.get("periodEnd");
}
```

Reply:
resolved in 9cfc864